src/wallet/wallet.cpp

```diff
@@ -3542,6 +3542,12 @@ void CWallet::LoadDescriptorScriptPubKeyMan(uint256 id, WalletDescriptor& desc)
         auto spk_manager = std::unique_ptr<ScriptPubKeyMan>(new DescriptorScriptPubKeyMan(*this, std::bind(&CWallet::CacheNewScriptPubKeys, this, std::placeholders::_1, std::placeholders::_2), desc, m_keypool_size));
         AddScriptPubKeyMan(id, std::move(spk_manager));
     }
+
+    // Cache scriptPubKeys
+    ScriptPubKeyMan* spkm = m_spk_managers.at(id).get();
+    for (const auto& script : spkm->GetScriptPubKeys()) {
+        m_cached_spks[script].insert(spkm);
+    }
```

Comment:
This does not seems to be necessary. At this point (`LoadDescriptorScriptPubKeyMan`), the spkm should't contain any script.

Reply:
Good catch, removed.